lighthouse-cli/test/smokehouse/frontends/lib.js

```diff
@@ -16,14 +16,48 @@
 const smokeTests = require('../test-definitions/core-tests.js');
 const {runSmokehouse} = require('../smokehouse.js');
 
+/**
+ * JSON.parse(JSON.stringify(object)), but persists RegExps too.
+ * @param {*} object
+ */
+function deepCopy(object) {
+  /**
+   * @param {string} _
+   * @param {*} value
+   */
+  function replacer(_, value) {
+    if (value instanceof RegExp) {
+      return ('__REGEXP ' + value.toString());
+    } else {
+      return value;
+    }
+  }
+
+  /**
+   * @param {string} _
+   * @param {*} value
+   */
+  function reviver(_, value) {
+    if (typeof value === 'string' && value.startsWith('__REGEXP ')) {
+      const m = value.split('__REGEXP ')[1].match(/\/(.*)\/(.*)?/);
+      if (!m) return value;
+      return new RegExp(m[1], m[2] || '');
+    } else {
+      return value;
+    }
+  }
+
+  return JSON.parse(JSON.stringify(object, replacer), reviver);
+}
+
 /**
  * @param {Smokehouse.SmokehouseLibOptions} options
  */
 async function smokehouse(options) {
   const {urlFilterRegex, skip, modify, ...smokehouseOptions} = options;
 
   /** @type {Smokehouse.TestDfn[]} */
-  const clonedTests = JSON.parse(JSON.stringify(smokeTests));
+  const clonedTests = deepCopy(smokeTests);
```

Comment:
this is just dev dependency why not use just use `_.cloneDeep`? :) https://lodash.com/docs/4.17.15#cloneDeep

Reply:
if we decide to keep and maintain this ourselves instead I expect rigorous tests and edge cases ;)

Reply:
sgtm